instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/_server.py

```diff
@@ -158,25 +154,34 @@ def set_code(self, code):
             SpanAttributes.RPC_GRPC_STATUS_CODE, code.value[0]
         )
         if code != grpc.StatusCode.OK:
-            self._active_span.set_status(
-                Status(
-                    status_code=StatusCode.ERROR,
-                    description=f"{code}:{details}",
-                )
-            )
+            status = self._server_status(code, details)
+            self._active_span.set_status(status)
         return self._servicer_context.set_code(code)
 
     def set_details(self, details):
         self._details = details
         if self._code != grpc.StatusCode.OK:
-            self._active_span.set_status(
-                Status(
-                    status_code=StatusCode.ERROR,
-                    description=f"{self._code}:{details}",
-                )
-            )
+            status = self._server_status(self._code, details)
+            self._active_span.set_status(status)
         return self._servicer_context.set_details(details)
 
+    def _server_status(self, code, details):
+        error_status = Status(
+            status_code=StatusCode.ERROR, description=f"{code}:{details}"
+        )
+        status_codes = {
```

Comment:
Thank you for putting in this PR that lines up with the [spec update](https://github.com/open-telemetry/opentelemetry-specification/pull/3333/files#diff-a4272fe55d12584e896526953d110354d9cc0c4ee26c6cdf559cbafdd50d9cea). The updated tests make sense.